marshal/consumer.go

```diff
@@ -639,8 +639,9 @@ func (c *Consumer) terminateAndCleanup(release bool, remove bool) bool {
 
 	latestTopicClaims := make(map[string]bool)
 	releasedTopics := make(map[string]bool)
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+
```

Comment:
Yeah it's unclear, the race stack doesn't have the reader so I'm making somewhat of a guess. The "write" was the closechan call in this method, so the read is probably using the channel in some way. I'm hoping this fixes it, but I don't know for sure since I don't know who race against it.